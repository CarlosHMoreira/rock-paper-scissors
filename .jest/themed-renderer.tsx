import React from 'react';
import { create } from 'react-test-renderer';

import Theme from '../src/theme';

export const renderWithTheme = (component: React.ReactNode) => create(<Theme>{component}</Theme>);