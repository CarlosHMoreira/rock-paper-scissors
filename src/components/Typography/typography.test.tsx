import React from 'react';
import 'jest-styled-components'

import { renderWithTheme } from '../../../.jest/themed-renderer';

import * as T from '.';

describe('Typography', () => {
  test('it should match the default title snapshot', () => {
    const component = (<T.Title>Rock paper scissors</T.Title>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match Title with different color snapshot', () => {
    const component = (<T.Title color="#fff" >Rock paper scissors</T.Title>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match Text default snapshot', () => {
    const component = (<T.Text>Rock paper scissors</T.Text>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match Text with different color snapshot', () => {
    const component = (<T.Text color="#fff">Rock paper scissors</T.Text>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('it should match Text with different color and bold snapshot', () => {
    const component = (<T.Text bold color="#fff">Rock paper scissors</T.Text>);
    const snapshot = renderWithTheme(component);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});