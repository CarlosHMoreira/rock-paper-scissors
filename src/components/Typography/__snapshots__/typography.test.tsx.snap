// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography it should match Text default snapshot 1`] = `
<p
  className=""
>
  Rock paper scissors
</p>
`;

exports[`Typography it should match Text with different color and bold snapshot 1`] = `
<p
  className=""
  color="#fff"
>
  Rock paper scissors
</p>
`;

exports[`Typography it should match Text with different color snapshot 1`] = `
<p
  className=""
  color="#fff"
>
  Rock paper scissors
</p>
`;

exports[`Typography it should match Title with different color snapshot 1`] = `
<h1
  className=""
  color="#fff"
>
  Rock paper scissors
</h1>
`;

exports[`Typography it should match the default title snapshot 1`] = `
<h1
  className=""
>
  Rock paper scissors
</h1>
`;
